bin/generate-ssl-options.sh

```diff
@@ -0,0 +1,23 @@
+#!/usr/bin/env bash
+#
+# Script to generate private RSA key and a self-signed certificate under src/main/resources/localhost-ssl (git-ignored)
+# These resources are used for exposing the application via HTTPS in development environment. They must not be used
+# in other environments, as their purpose is to provide consistency, not security.
+
+localhost_ssh_folder="src/main/resources/localhost-ssl"
+
+if [ ! -f "$localhost_ssh_folder"/localhost.key -o ! -f "$localhost_ssh_folder"/localhost.crt ]
+then
+  mkdir -p "$localhost_ssh_folder"
+  openssl req \
+    -nodes \
+    -x509 \
+    -newkey rsa:4096 \
+    -keyout "$localhost_ssh_folder"/localhost.key \
+    -out "$localhost_ssh_folder"/localhost.crt \
+    -sha256 \
+    -days 3650 \
+    -subj "/C=GB/ST=A/L=B/O=C/OU=D/CN=E" \
+    -addext "subjectAltName = DNS:hmcts.net,IP:127.0.0.1,IP:198.168.1.1" \
```

Comment:
unfortunately default mac os openssl doesn't have this option, the second part of the answer explains how to solve this.

we want this to work out of the box for developers as much as possible.

Should probably detect if it's mac os to that by default we use the up to date command?

```
➜  expressjs-template git:(refactor/remove_localhost_private_key_from_vcs) uname
Darwin
```
`uname == "Darwin"` should work as a check

Reply:
@timja I've changed the script to check for macOS. Can you review and test this for me please? My local setup is a linux VM hosted by Windows